File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {

        map <int,int> mapp;
        for(auto x:nums1)mapp[x]++;
        vector <int> ans;
        for(auto x:nums2){
            if(mapp[x]>0){
                mapp[x]--;
                ans.push_back(x);
            }
        }
        return ans;
    }
};

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:

        mp = defaultdict(int)
        for x in nums1:mp[x]+=1

        ans = []
        for x in nums2:
            val = mp.get(x,0)
            if val>0:
                ans.append(x)
                mp[x]-=1
        
        return ans

        